core/base_abstractions/sensor.py

```diff
@@ -158,23 +166,32 @@ def _get_observation_space(self) -> gym.spaces.Tuple:
         only if the expert failed to generate a true expert action. The
         value `num actions in task` should be in `config["nactions"]`
         """
-        return gym.spaces.Tuple(
-            (gym.spaces.Discrete(self.nactions), gym.spaces.Discrete(2))
-        )
+        return flatten_space(self.unflattened_observation_space)
 
     def get_observation(
         self, env: EnvType, task: SubTaskType, *args: Any, **kwargs: Any
     ) -> Any:
         # If the task is completed, we needn't (perhaps can't) find the expert
         # action from the (current) terminal state.
         if task.is_done():
-            return np.array([-1, False], dtype=np.int64)
+            return np.array([self.action_space.sample(), False])
+
         action, expert_was_successful = task.query_expert(**self.expert_args)
-        assert isinstance(action, int), (
-            "In expert action sensor, `task.query_expert()` "
-            "did not return an integer action."
+
+        if isinstance(action, int):
+            assert isinstance(self.action_space, gym.spaces.Discrete)
+            unflattened_action = action
+        else:
+            # Assume we receive a gym-flattened numpy action
+            unflattened_action = gym_unflatten(self.action_space, action)
+
+        unflattened_torch = torch_point(
+            self.unflattened_observation_space,
+            (unflattened_action, expert_was_successful),
         )
-        return np.array([action, expert_was_successful], dtype=np.int64)
+
+        flattened_torch = flatten(self.unflattened_observation_space, unflattened_torch)
+        return flattened_torch.cpu().numpy()
```

Comment:
This seems like a reasonable solution to me 👍

It does make me wonder though: I originally appended the `expert_was_successful` bool to the end of the tensor as a workaround as our rollout storage wasn't equipped to deal with anything other than tensors. Maybe the harder but "right" solution is to have this sensor return a `Dict` with a form something like
```python
{
    "action": ...,
    "was_found": ....
}
```
People would then query the observation as `observation["expert_action"]["action"]`. Thoughts?

Reply:
Since all this is managed by the engine, I would leave it like this for now... In any case, if you're really concerned about the perhaps hacky current solution, feel free to insist on this and I'll take care :)